libraries/ESP8266WiFi/src/include/UdpContext.h

```diff
@@ -522,6 +528,90 @@ class UdpContext
         reinterpret_cast<UdpContext*>(arg)->_recv(upcb, p, srcaddr, srcport);
     }
 
+#if LWIP_VERSION_MAJOR == 1
+    /*
+     * Code in this conditional block is copied/backported verbatim from
+     * LwIP 2.1.2 to provide pbuf_get_contiguous.
+     */
+
+    static const struct pbuf *
```

Comment:
It looks to me like this method doesn't have to be in the class definition. I suggest moving to the .cpp, maybe even move it out of the class altogether.
Of course, the other methods that make use of it should be moved as well.

Reply:
There is no cpp. UdpContext is a header-only class. These 2 methods are used exclusively in this class, so I believe it's better to enclose it inside and not pollute the global namespace.

Reply:
I meant create one and move this in there. But there is additional cleanup to do here, so let's just leave this as is.